app/src/org/commcare/activities/QueryRequestActivity.java

```diff
@@ -142,16 +142,22 @@ private void buildPromptEntry(LinearLayout promptsLayout, String promptId,
         View promptView = LayoutInflater.from(this).inflate(R.layout.query_prompt_layout, promptsLayout, false);
         setLabelText(promptView, queryPrompt.getDisplay());
         View inputView;
-        String input = queryPrompt.getInput();
-        if (input != null && input.contentEquals(INPUT_TYPE_SELECT1)) {
-            inputView = buildSpinnerView(promptView, queryPrompt);
-        } else {
-            inputView = buildEditTextView(promptView, queryPrompt, isLastPrompt);
+        if (isPromptValid(queryPrompt)) {
+            String input = queryPrompt.getInput();
+            if (input != null && input.contentEquals(INPUT_TYPE_SELECT1)) {
+                inputView = buildSpinnerView(promptView, queryPrompt);
+            } else {
+                inputView = buildEditTextView(promptView, queryPrompt, isLastPrompt);
+            }
+            setUpBarCodeScanButton(promptView, promptId, queryPrompt);
+
+            promptsLayout.addView(promptView);
+            promptsBoxes.put(promptId, inputView);
         }
-        setUpBarCodeScanButton(promptView, promptId, queryPrompt);
+    }
 
-        promptsLayout.addView(promptView);
-        promptsBoxes.put(promptId, inputView);
+    private boolean isPromptValid(QueryPrompt queryPrompt) {
```

Comment:
Nit: I would find 'valid' to be a bit confusing if I ran across this code. I might rename this as something like 'isPromptSupported'

Reply:
```suggestion
    private boolean isPromptSupported(QueryPrompt queryPrompt) {
```